benchmarks/__main__.py

```diff
@@ -1,27 +1,60 @@
 """Executes the benchmarking module."""
 # Run this via 'python -m benchmarks' from the root directory.
 
+import argparse
 import os
 
+from benchmarks.definition import Benchmark
 from benchmarks.domains import BENCHMARKS
 from benchmarks.persistence import (
     LocalFileObjectStorage,
     PathConstructor,
     S3ObjectStorage,
 )
+from benchmarks.result import Result
 
 RUNS_IN_CI = "CI" in os.environ
 
 
-def main():
+def save_benchmark_data(benchmark: Benchmark, result: Result) -> None:
+    """Save the benchmark data to the object storage."""
+    path_constructor = PathConstructor.from_result(result)
+    persist_dict = benchmark.to_dict() | result.to_dict()
+
+    object_storage = S3ObjectStorage() if RUNS_IN_CI else LocalFileObjectStorage()
+    object_storage.write_json(persist_dict, path_constructor)
+
+
+def run_all_benchmarks() -> None:
     """Run all benchmarks."""
     for benchmark in BENCHMARKS:
         result = benchmark()
-        path_constructor = PathConstructor.from_result(result)
-        persist_dict = benchmark.to_dict() | result.to_dict()
+        save_benchmark_data(benchmark, result)
+
 
-        object_storage = S3ObjectStorage() if RUNS_IN_CI else LocalFileObjectStorage()
-        object_storage.write_json(persist_dict, path_constructor)
+def run_specific_benchmarks(benchmark_subset_names: set[str]) -> None:
```

Comment:
Would change the signature slightly (body needs to change accordingly, of course)
```suggestion
def run_benchmarks(benchmark_names: Collection[str]) -> None:
```